cmd/cstor-pool-mgmt/controller/new-pool-controller/handler.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package poolcontroller
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-pool-mgmt/controller/common"
+
+	zpool "github.com/openebs/maya/cmd/cstor-pool-mgmt/pool/v1alpha2"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	apis2 "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha2"
+	corev1 "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// reconcile will ensure that pool for given
+// key is created and running
+func (c *CStorPoolController) reconcile(key string) error {
+	var err error
+	var isImported bool
+
+	csp, err := c.getCSPObjFromKey(key)
+	if err != nil {
+		return err
+	}
+
+	if zpool.IsReconcileDisabled(csp) {
```

Comment:
this as well need a revisit.. zpool package having k8s related things doesn't sounds correct

Reply:
many such functions are there

Reply:
yes. We can move this to the controller itself since all such functions are being called from controller only.